packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -767,14 +767,13 @@ function lastMessageFromMetadata(metadata: IContainerRuntimeMetadata | undefined
  * We only want to log this once, to avoid spamming telemetry if we are wrong and these cases are hit commonly.
  */
 let getSingleUseLegacyLogCallback = (logger: ITelemetryLoggerExt, type: string) => {
-	// We only want to log this once per ContainerRuntime instance, to avoid spamming telemetry.
-	getSingleUseLegacyLogCallback = () => () => {};
-
 	return (codePath: string) => {
 		logger.sendTelemetryEvent({
 			eventName: "LegacyMessageFormat",
 			details: { codePath, type },
 		});
+		// We only want to log this once per ContainerRuntime instance, to avoid spamming telemetry.
+		getSingleUseLegacyLogCallback = () => () => {};
```

Comment:
Thanks, good fix

Reply:
Put this in a separate PR with Unit Test coverage:  https://github.com/microsoft/FluidFramework/pull/22303